packages/headless/src/features/recent-results/recent-results-slice.test.ts

```diff
@@ -0,0 +1,95 @@
+import {buildMockResult} from '../../test';
+import {
+  clearRecentResults,
+  pushRecentResult,
+  registerRecentResults,
+} from './recent-results-actions';
+import {recentResultsReducer} from './recent-results-slice';
+import {
+  getRecentResultsInitialState,
+  RecentResultsState,
+} from './recent-results-state';
+
+describe('recent-results slice', () => {
+  let state: RecentResultsState;
+
+  const resultStringParams = {
+    title: 'title',
+    uri: 'uri',
+    printableUri: 'printable-uri',
+    clickUri: 'click-uri',
+    uniqueId: 'unique-id',
+    excerpt: 'exceprt',
+    firstSentences: 'first-sentences',
+    flags: 'flags',
+  };
+
+  const mockResult = buildMockResult(resultStringParams);
+  const otherMockResult = buildMockResult({
+    ...resultStringParams,
+    uniqueId: 'different-id-1',
+  });
+  const mockResults = [mockResult];
+  const testMaxLength = 5;
+
+  beforeEach(() => {
+    state = getRecentResultsInitialState();
+  });
+
+  it('should have initial state', () => {
+    expect(recentResultsReducer(undefined, {type: 'foo'})).toEqual({
+      maxLength: 10,
+      results: [],
+    });
+  });
+
+  it('#registerRecentResults should set results and maxLength params in state', () => {
+    state = recentResultsReducer(
+      state,
+      registerRecentResults({results: mockResults, maxLength: testMaxLength})
+    );
+
+    expect(state.results).toEqual(mockResults);
+    expect(state.maxLength).toEqual(testMaxLength);
+  });
+
+  it('#clearRecentResults should remove all results from the queue in state', () => {
+    state.results = mockResults;
+
+    state = recentResultsReducer(state, clearRecentResults());
+
+    expect(state.results).toEqual([]);
+  });
+
+  it('#pushRecentResult should add new recent result if queue is empty', () => {
+    state = recentResultsReducer(state, pushRecentResult(mockResult));
+
+    expect(state.results).toEqual([mockResult]);
+  });
+
+  it('#pushRecentResult should add new recent result if queue is non-empty', () => {
+    state.results = [otherMockResult];
+    state = recentResultsReducer(state, pushRecentResult(mockResult));
+
+    expect(state.results.length).toBe(2);
+  });
```

Comment:
Is result order important ? If so, we should probably check element order in this test.